release_checklist.md

```diff
@@ -1,47 +1,117 @@
+* Upload and test the pre-release Python package.
+    1) `python setup.py build sdist`
+    1) `twine upload dist/whatever.tar.gz`  (Needs credentials on pypi)
+    1) ON ANOTHER MACHINE AND CHECK THAT THE VERSION IS CORRECT AND IT ACTUALLY IS BUILDING THINGS IN THE LOG: `sudo pip install jsonnet --pre --upgrade`
+    1) `python`
+    1) `import _jsonnet`
+    1) `_jsonnet.evaluate_snippet('foo', '1+1')`
+* Remove -pre from version in `include/libjsonnet.h`.
+* Update version in `CMakeLists.txt`.
+* Update `test_cmd` golden file version numbers:
 ```
-1) Check if any changes to the documentation are necessary by checking commits since previous release (especially stdlib additions).
-1) In stdlib documentation replace any "Available in upcoming release." with "Available since NEW_VERSION_GOES_HERE".
-1) `python setup.py build sdist`
-1) `twine upload dist/whatever.tar.gz`  (Needs credentials on pypi)
-1) ON ANOTHER MACHINE AND CHECK THAT THE VERSION IS CORRECT AND IT ACTUALLY IS BUILDING THINGS IN THE LOG: `sudo pip install jsonnet --pre --upgrade`
-1) `python`
-1) `import _jsonnet`
-1) `_jsonnet.evaluate_snippet('foo', '1+1')`
-1) Checkout go-jsonnet master
-1) update cpp-jsonnet commit
-1) `./tests`
-
-## Do release
-1) Remove -pre from version in `include/libjsonnet.h`
-1) Update version in `CMakeLists.txt`
-1) Update `test_cmd` golden file version numbers
-1) run test again!
-1) commit and push
-1) Make release in github, write release notes by checking commits since previous release
-1) `python setup.py build sdist`
-1) `twine upload dist/whatever.tar.gz` (the version without the -pre)
-1) Post to Jsonnet Google Group
-1) Update the live version of the website
+find test_cmd -name '*.cpp' -o -name '*.golang' -o -name '*.stdout' -o -name '*.stderr' | \
+xargs sed -i 's/ v0[.][0-9.]*/ NEW_VERSION_GOES_HERE/g'
+```
+* In stdlib documentation replace any "Available in upcoming release." with "Available since NEW_VERSION_GOES_HERE".
+* Send a PR and wait for CI to be green before merging.
+* Create a release on Github. Put the previously prepared release notes there.
+* Upload the Python package.
+    1) `python setup.py build sdist`
+    1) `twine upload dist/whatever.tar.gz` (the version without the -pre)
+
+## Release the Go version
+
+* Sync with the (freshly-released) C++ version.
+* At this point the C++ tests will not pass due to difference in version.
+* Modify `vm.go` to bump the version number, but add `-pre1` suffix.
+* Upload and test the pre-release Python package.
+    1) `python setup.py build sdist`
+    1) `twine upload dist/whatever.tar.gz`  (Needs credentials on pypi)
+    1) ON ANOTHER MACHINE AND CHECK THAT THE VERSION IS CORRECT AND IT ACTUALLY IS BUILDING THINGS IN THE LOG: `sudo pip install gojsonnet --pre --upgrade`
+    1) `python`
+    1) `import _gojsonnet`
+    1) `_gojsonnet.evaluate_snippet('foo', '1+1')`
+* Remove the `-pre1` from version in `vm.go`. Now the tests should pass again.
```

Comment:
Possible source of confusion – running the tests before commiting results in an accidental rollback of cpp-jsonnet.

Reply:
No longer an issue, because the confusing behavior was removed from tests.sh.